text/0069-typed-event.md

```diff
@@ -0,0 +1,150 @@
+
+## Internal details
+
+This section specifies data format in Protocol Buffers for `TimedEvent` messages within the overall OTLP. It follows and expands on the flattened data structure in [#59 OTLP Trace Data Format](https://github.com/open-telemetry/oteps/pull/59).
+
+### Resource
+
+```protobuf
+  // TimedEvent is a time-stamped annotation of the span, consisting of either
+  // user-supplied key-value pairs, details of a message sent/received between Spans or
+  // information about an error, fault or exception
+  message TimedEvent {
+    enum Type {
+      // Unknown event type.
+      TYPE_UNSPECIFIED = 0;
+      // Contains only timestamp and attributes.
+      ANNOTATION = 1;
+      // Indicates a sent message.
+      MESSAGE_SENT = 2;
+      // Indicates a received message.
+      MESSAGE_RECEIVED = 3;
+      // Indicates an error, fault or exception occurred.
+      ERROR = 4;
+    }
+
+    // The type of MessageEvent. Indicates whether the message was sent or
+    // received.
+    Type type = 1;
+
+    // time_unixnano is the time the event occurred.
+    int64 time_unixnano = 2;
+
+    // name is a user-supplied description of the event.
+    string name = 3;
+
+    // attributes is a collection of attribute key/value pairs on the event.
+    repeated AttributeKeyValue attributes = 4;
+
+    // dropped_attributes_count is the number of dropped attributes. If the value is 0,
+    // then no attributes were dropped.
+    int32 dropped_attributes_count = 5;
+
+    //// Fields for use only by MESSAGE_SENT and MESSAGE_RECEIVED ////
+
+    // An identifier for the MessageEvent's message that can be used to match
+    // SENT and RECEIVED MessageEvents. For example, this field could
+    // represent a sequence ID for a streaming RPC. It is recommended to be
+    // unique within a Span.
+    uint64 message_id = 6;
+
+    // The number of uncompressed bytes sent or received.
+    uint64 uncompressed_size = 7;
+
+    // The number of compressed bytes sent or received. If zero, assumed to
+    // be the same size as uncompressed.
+    uint64 compressed_size = 8;
+
+    // The content or body of the message.
+    bytes message_content = 9;
+
+    //// Fields for use only by ERROR ////
+
+    message Exception {
+      // Unique identifier within a parent span for the exception.
+      bytes id = 1;
+      // The exception message.
+      string messsage = 2;
+      // The exception class or type.
+      string type = 3;
+      // Exception ID of the exception's parent, that is, the exception that caused this exception.
+      bytes cause = 4;
+      // The stack.
+      StackTrace stack = 5;
+    }
+
+    // Collection of exceptions which triggered the error or fault.
```

Comment:
It's not clear what a "collection" means semantically. Usually exceptions are nested.

Reply:
Good point. I just copied the AWS X-Ray approach with parent ids denoting nesting. I will look at improving comprehensibility.